transport/http2_client.go

```diff
@@ -209,6 +210,8 @@ func newHTTP2Client(ctx context.Context, addr TargetInfo, opts ConnectOptions) (
 		goAway:            make(chan struct{}),
 		awakenKeepalive:   make(chan struct{}, 1),
 		framer:            newFramer(conn),
+		hBuf:              &buf,
```

Comment:
`hBuf: &bytes.Buffer{}`

Reply:
I know we talked about this but now I realize that hEnc needs the same buffer.

Reply:
Yes, that should have been obvious, sorry!